pkg/speaker/vip/keepalived.go

```diff
@@ -350,3 +355,23 @@ func newKeepalivedLogPiper() io.WriteCloser {
 	}()
 	return writer
 }
+
+// Stop stop keepalived process
+func (k *keepAlived) Stop() {
+	klog.Infof("Cleanup: %s", k.vips)
+	for _, i := range k.instances {
+		klog.Infof("removing configured VIP %v", i.Svcips)
+		for _, vip := range i.Svcips {
+			out, err := k8sexec.New().Command("ip", "addr", "del", vip+"/32", "dev", i.Iface).CombinedOutput()
```

Comment:
Why is there a need to actively delete the vip here?

My understanding is that it should be automatically deleted after keepAlived stops, right?

Reply:
> Why is there a need to actively delete the vip here?
> 
> My understanding is that it should be automatically deleted after keepAlived stops, right?

You’ve understood correctly. The current issue is that when the main process exits, the keepalived process of the speaker has not been fully reclaimed, causing the problem.